src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -92,6 +112,45 @@ public void setPerson(Person target, Person editedPerson) {
      */
     public void removePerson(Person key) {
         persons.remove(key);
+        tasks.remove(key.getTask());
```

Comment:
this line would throw an NullPointerException if the person do not have any task.

```suggestion
        Task task = key.getTask();
        if (task != null) {
            tasks.remove(key.getTask());
        }
```